packages/json-rpc-engine/src/v2/JsonRpcEngineV2.ts

```diff
@@ -0,0 +1,309 @@
+import {
+  type Json,
+  type JsonRpcRequest,
+  type JsonRpcNotification,
+  type NonEmptyArray,
+  hasProperty,
+} from '@metamask/utils';
+import deepFreeze from 'deep-freeze-strict';
+
+import { MiddlewareContext } from './MiddlewareContext';
+import {
+  isNotification,
+  isRequest,
+  JsonRpcEngineError,
+  stringify,
+} from './utils';
+import type { JsonRpcCall } from './utils';
+
+export type Next<Request extends JsonRpcCall, Result extends Json | void> = (
+  request?: Readonly<Request>,
```

Comment:
Thoughts on making this required? It's a little more concise to leave it out when you're not substituting the request with another, but on the other hand, requiring it would eliminate a branch and would make it more "obvious" to the reader/writer what is happening and how to modify the request if necessary.

Reply:
Hm, not a bad idea. I don't think it would eliminate a branch, because we'd still check if the request changed to avoid trying to deep-freeze it again (see [here](https://github.com/MetaMask/core/blob/42264e24010b24548c3b39be3990a9bc9b1b11eb/packages/json-rpc-engine/src/v2/JsonRpcEngineV2.ts#L191-L194)), but would still be a clarity gain I think.

Reply:
The branch it would eliminate is this one: https://github.com/MetaMask/core/blob/42264e24010b24548c3b39be3990a9bc9b1b11eb/packages/json-rpc-engine/src/v2/JsonRpcEngineV2.ts#L182
(the assigning of a default value). It's not counted by our code coverage tools as a branch, but I still think of it as one.

Just clarifying, not that one default parameter value really matters at all of course.

Reply:
Ah of course, got it.

Reply:
The decision for now is to not make passing the request required, but we could be swayed by strong arguments to the contrary.